src/helpers/request.ts

```diff
@@ -65,7 +55,7 @@ type RequestGenerator<Args, ResponseType> = {
 };
 
 type GeneratedRequestFunction<Args, ResponseType> = (
-  ...a: Parameters<HandleRequest<Args>>
+  ...a: Parameters<RequestGenerator<Args, ResponseType>['handleRequest']>
```

Comment:
What was the reason for the change on this line? I'm not sure I see how it connects to the other changes in this PR.

Reply:
This change was me trying to debug the fact that `unsplash.photos.get()` didn't work. Making this type reuse `RequestGenerator` instead of `HandleRequest` directly meant the type were more consolidated, as I only had to tweak the types in `RequestGenerator` to see if i can get it to work.

Decided to leave this change as it more explicitly shows what we're doing: `GeneratedRequestFunction` is always outputting whatever parameters `RequestGenerator['handleRequest']` has.

Reply:
I can see how it may have helped for debugging purposes but I'm not sure I can see how it is more consolidated/explicit—but maybe it's subjective. I just felt like the new version takes more mental steps to read compared to just using `HandleRequest`.

Reply:
I don't mind too much either way, so I reverted 👍